packages/plugin/vite/src/VitePlugin.ts

```diff
@@ -116,14 +115,7 @@ the generated files). Instead, it is ${JSON.stringify(pj.main)}`);
       delete pj.config.forge;
     }
 
-    await fs.writeJson(path.resolve(buildPath, 'package.json'), pj, {
-      spaces: 2,
-    });
-
-    // Copy the dependencies in package.json
-    for (const dep of flatDependencies) {
-      await fs.copy(dep.src, path.resolve(buildPath, dep.dest));
-    }
```

Comment:
Am I reading this correctly and this removes the functionality to copy dependencies listed in `package.json`? Is this supposed to land in a forge v8 package or will this breaking change land in a minor version?

Reply:
This should be released as a minor version, because Vite does not build some Node.js modules reliably.
However, most people still want Vite to handle the Node.js modules in `dependencies`, which can effectively reduce the size of the App build.

Reply:
> In any case, the current change is the choice of most people.

And how has this been measured?

> If other problems are encountered during the upgrade process, we may need some community-provided Vite plugins to handle them instead of forge.

Do you mean a vite plugin to include the native binaries in the bundle?

In any case why is it _necessary_ to break existing configuration now? AFAICT it would be possible to add an option to toggle this behavior.

Reply:
In general, people who have used `@electron-forge/plugin-webpack` plugin hope that **vite-plugin** can have the same design as **webpack-plugin**, so that they can easily migrate from Webpack to Vite. But in fact, Vite and Webpack are not compatible in many places, and we can only work hard to make it more like Webpack. At least the current PR behavior is consistent with Webpack.
We may need to remove some hack behaviors from Forge, and they will be done by Vite plugin. For example, the copy behavior of `dependencies` looks very stupid in the eyes of Webpack users, but Vite often cannot solve such problems, especially C/C++ native packages.
Of course, I personally think that the current version of Vite plugin is the result of weighing many aspects, but it is not easy to be accepted by Webpack plugin users.

Reply:
For users, the upgrade will bring some additional problems, which is not only a problem faced by Forge users, but more accurately, it should be a problem that Vite needs to face. I will try my best to provide some additional plugins to help users solve these problems, and also to provide Vite ecosystem with more Webpack-like capabilities.

Reply:
> In any case why is it necessary to break existing configuration now? AFAICT it would be possible to add an option to toggle this behavior.

Adding an option to let the user decide whether to copy `dependencies` would be necessary, as this PR is clearly a departure from previous behavior.
Please let me know if anyone agrees to add this option?